lib/irc.js

```diff
@@ -517,18 +515,29 @@ function Client(server, nick, opt) {
 
             // for sasl
             case "CAP":
-                if ( message.args[0] === '*' &&
+              if ( self.opt.debug ) {
+                util.log(message.args[0]+"-")
+                util.log(message.args[1]+"-")
+                util.log(message.args[2]+"-")
+              }
+                if ( //message.args[0] === '*' &&
                      message.args[1] === 'ACK' &&
-                     message.args[2] === 'sasl ' ) // there's a space after sasl
+                     message.args[2] === 'sasl' ) { // there's a space after sasl
+                     self.send("NICK", self.opt.nick);
                     self.send("AUTHENTICATE", "PLAIN");
-                break;
+
+                    break;
+                  }
             case "AUTHENTICATE":
-                if ( message.args[0] === '+' ) self.send("AUTHENTICATE",
-                    new Buffer(
-                        self.opt.nick + '\0' +
-                        self.opt.userName + '\0' +
-                        self.opt.password
-                    ).toString('base64'));
+                if ( message.args[0] === '+' )
+                  var buf = new Buffer(
+                      self.opt.nick + '\0' +
+                      self.opt.saslUserName + '\0' +
```

Comment:
I'd much prefer if you checked for `self.opt.saslUserName` in the settings before using it. If it isn't defined, stick with the original behavior. This would let it remain backwards compatible with existing configurations.